vault/vault.go

```diff
@@ -227,6 +237,8 @@ func (t *tempCredsCreator) provider(config *Config) (credentials.Provider, error
 		return NewSSORoleCredentialsProvider(t.keyring.Keyring, config)
 	} else if config.HasRole() && (config.HasWebIdentityTokenFile() || config.HasWebIdentityTokenProcess()) {
 		return NewAssumeRoleWithWebIdentityProvider(t.keyring.Keyring, config)
+	} else if config.HasCredentialProcess() {
+		return NewCredentialProcessProvider(config)
```

Comment:
This is the source credentials right? So should probably be `sourceCredProvider = NewCredentialProcessProvider(config)`

Reply:
What are "master credentials" ? Are they used directly, without creating a temporary session like the `--no-session` parameter ?